File: CS-Examples/09_Charts/CreateHistogramChart.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls

Namespace CreateHistogramChart
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the Excel file "HistogramChart.xlsx" from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\HistogramChart.xlsx")

            ' Get the first worksheet from the workbook
            Dim sheet = workbook.Worksheets(0)

            ' Add a chart to the worksheet
            Dim officeChart = sheet.Charts.Add()

            ' Set the chart type to Histogram
            officeChart.ChartType = ExcelChartType.Histogram

            ' Set the data range for the chart
            officeChart.DataRange = sheet("A1:A15")

            ' Set the position of the chart within the worksheet using row and column indices
            officeChart.TopRow = 1
            officeChart.BottomRow = 19
            officeChart.LeftColumn = 4
            officeChart.RightColumn = 12

            ' Set the bin width for the primary category axis (X-axis)
            officeChart.PrimaryCategoryAxis.BinWidth = 8

            ' Set the gap width between bars in the chart series
            officeChart.Series(0).DataFormat.Options.GapWidth = 6

            ' Set the chart title to "Height Data"
            officeChart.ChartTitle = "Height Data"

            ' Set the title for the primary value axis (Y-axis)
            officeChart.PrimaryValueAxis.Title = "Number of students"

            ' Set the title for the primary category axis (X-axis)
            officeChart.PrimaryCategoryAxis.Title = "Height"

            ' Disable the legend in the chart
            officeChart.HasLegend = False

            ' Save the modified workbook to "Histogram_chart.xlsx"
            workbook.SaveToFile("Histogram_chart.xlsx")
            ' Release the resources used by the workbook
            workbook.Dispose()

            ExcelDocViewer("Histogram_chart.xlsx")
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub


	End Class
End Namespace

File: CS-Examples/09_Charts/CreateHistogramChart.cs
using System;
using System.Windows.Forms;
using Spire.Xls;

namespace CreateHistogramChart
{
	public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
		private void btnRun_Click(object sender, System.EventArgs e)
		{
            //Create a new excel document
            Workbook workbook = new Workbook();

            //Load an excel document from the file
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\HistogramChart.xlsx");

            //Find the first worksheet
            var sheet = workbook.Worksheets[0];

            //Add a new chart
            var officeChart = sheet.Charts.Add();

            //Set chart type as histogram       
            officeChart.ChartType = ExcelChartType.Histogram;

            //Set data range in the worksheet   
            officeChart.DataRange = sheet["A1:A15"];
            officeChart.TopRow = 1;
            officeChart.BottomRow = 19;
            officeChart.LeftColumn = 4;
            officeChart.RightColumn = 12;

            //Category axis bin settings        
            officeChart.PrimaryCategoryAxis.BinWidth = 8;

            //Gap width settings
            officeChart.Series[0].DataFormat.Options.GapWidth = 6;

            //Set the chart title and axis title
            officeChart.ChartTitle = "Height Data";
            officeChart.PrimaryValueAxis.Title = "Number of students";
            officeChart.PrimaryCategoryAxis.Title = "Height";

            //Hiding the legend
            officeChart.HasLegend = false;

            // Save the file 
            workbook.SaveToFile("Histogram_chart.xlsx");

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the file 
            ExcelDocViewer("Histogram_chart.xlsx");
		}

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }


	}
}
